include/boost/beast/core/detail/impl/any_buffers.hpp

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2016-2021 Vinnie Falco (vinnie dot falco at gmail dot com)
+// Copyright (c) 2021 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+template<bool IsConst>
+template<class Buffers, class Alloc>
+void any_buffers<IsConst>::impl<Buffers, Alloc>::destroy() noexcept
+{
+    using allocator_type = typename
+        std::allocator_traits<Alloc>::template
+            rebind_alloc<impl>;
+    auto alloc = allocator_type(a_);
+    using atraits = std::allocator_traits<allocator_type>;
+    atraits::destroy(alloc, this);
+    atraits::deallocate(alloc, this);
+}
+
+// ------------------------------------------------------------
+
+template<bool IsConst>
+template<class Buffers, class Alloc>
+static
+auto
+any_buffers<IsConst>::construct(Buffers const& b, Alloc const& a)
+-> impl_base*
+{
+    using impl_type = impl<Buffers, Alloc>;
+    using allocator_type = typename std::allocator_traits<Alloc>::template rebind_alloc<impl_type>;
+    auto alloc = allocator_type(a);
+    using atraits = std::allocator_traits<allocator_type>;
+
+    auto p = atraits::allocate(a, sizeof(impl_type));
```

Comment:
`sizeof(impl_type)` needs to be something like `sizeof(impl_type) + sizeof(value_type) * std::distance(net::buffer_sequence_begin(b), net::buffer_sequence_end(b))`

Reply:
yes, that's going in next.
Since this object never moves or copies I can actually make it variable size at construction by having the last member as a 1-array of value_type.